docs/Parameters-Tuning.rst

```diff
@@ -36,15 +36,143 @@ To get good results using a leaf-wise tree, these are some important parameters:
 For Faster Speed
 ----------------
 
--  Use bagging by setting ``bagging_fraction`` and ``bagging_freq``
+Add More Computational Resources
+''''''''''''''''''''''''''''''''
 
--  Use feature sub-sampling by setting ``feature_fraction``
+On systems where it is available, LightGBM uses OpenMP to parallelize many operations. The maximum number of threads used by LightGBM is controlled by the parameter ``num_threads``. By default, this will defer to the default behavior of OpenMP (one thread per real CPU core or the value in environment variable ``OMP_NUM_THREADS``, if it is set). For best performance, set this to the number of real CPU cores available.
+
+You might be able to achieve faster training by moving to a machine with more available CPU cores.
+
+Using distributed (multi-machine) training might also reduce training time. See the `Distributed Learning Guide <./Parallel-Learning-Guide.rst>`_ for details.
+
+Use a GPU-enabled version of LightGBM
+'''''''''''''''''''''''''''''''''''''
+
+You might find that training is faster using a GPU-enabled build of LightGBM. See the `GPU Tutorial <./GPU-Tutorial.rst>`__ for details.
+
+Grow Shallower Trees
+''''''''''''''''''''
+
+The total training time for LightGBM increases with the total number of tree nodes added. LightGBM comes with several parameters that can be used to control the number of nodes per tree.
+
+The suggestions below will speed up training, but might hurt statistical performance.
```

Comment:
Maybe replace "statistical performance" with "training accuracy"? "Performance" is often used to mean "speed" so I think this would be less ambiguous. Also it might be good to mention that these options will also reduce overfitting, which can improve accuracy on the validation set.

Reply:
oh good point! Ok I like the idea to use the term "training accuracy".

> Also it might be good to mention that these options will also reduce overfitting

I actually intentionally chose NOT to say that. There is a separate section in this document on how to improve training accuracy and another on how to fight overfitting. I wanted to keep this section only focused on runtime. I think being very straightforward like that makes it easier for people to reason about the direction and size of effects from changing these parameters.

The docs at https://lightgbm.readthedocs.io/en/latest/Parameters.html do have entries under each parameter like "can be used to reduce overfitting", "can be used to speed up training", etc.